### PR TITLE
FEATURE: Merge discourse-automation

### PR DESCRIPTION
**Pull request by OsamaSayegh**

### CODE REVIEW

**Review by OsamaSayegh — COMMENTED** [review]
To make it easy for reviewers to review this PR, I've made all new changes to core and the plugin in a single commit: https://github.com/discourse/discourse/pull/26432/commits/555de34aabe81a611e4862afab4eae8b8c4d69ab. Everything else is just the [plugin code](https://github.com/discourse/discourse-automation) copied over to core.

File: app/models/topic_custom_field.rb [hunk 1]
```diff
@@ -19,6 +19,8 @@ class TopicCustomField < ActiveRecord::Base
 #
 # Indexes
 #
-#  index_topic_custom_fields_on_topic_id_and_name  (topic_id,name)
-#  topic_custom_fields_value_key_idx               (value,name) WHERE ((value IS NOT NULL) AND (char_length(value) < 400))
+#  idx_topic_custom_fields_auto_responder_triggered_ids_partial    (topic_id,value) UNIQUE WHERE ((name)::text = 'auto_responder_triggered_ids'::text)
+#  idx_topic_custom_fields_discourse_automation_unique_id_partial  (topic_id,value) UNIQUE WHERE ((name)::text = 'discourse_automation_ids'::text)
```
Comment (tgxworld) [inline app/models/topic_custom_field.rb:23]: Same concern as https://github.com/discourse/discourse/pull/26432/files#r1548913457

File: app/models/user_custom_field.rb [hunk 2]
```diff
@@ -26,5 +26,6 @@ class UserCustomField < ActiveRecord::Base
 #
 # Indexes
 #
-#  index_user_custom_fields_on_user_id_and_name  (user_id,name)
+#  idx_user_custom_fields_discourse_automation_unique_id_partial  (user_id,value) UNIQUE WHERE ((name)::text = 'discourse_automation_ids'::text)
```
Comment (tgxworld) [inline app/models/user_custom_field.rb:29]: Ditto https://github.com/discourse/discourse/pull/26432/files#r1548913457

**Review by tgxworld — APPROVED** [review]

File: app/models/post_custom_field.rb [hunk 3]
```diff
@@ -19,9 +19,11 @@ class PostCustomField < ActiveRecord::Base
 #
 # Indexes
 #
-#  index_post_custom_fields_on_name_and_value    (name, "left"(value, 200))
-#  index_post_custom_fields_on_notice            (post_id) UNIQUE WHERE ((name)::text = 'notice'::text)
-#  index_post_custom_fields_on_post_id           (post_id) UNIQUE WHERE ((name)::text = 'missing uploads'::text)
-#  index_post_custom_fields_on_post_id_and_name  (post_id,name)
-#  index_post_id_where_missing_uploads_ignored   (post_id) UNIQUE WHERE ((name)::text = 'missing uploads ignored'::text)
+#  idx_post_custom_fields_discourse_automation_unique_id_partial  (post_id,value) UNIQUE WHERE ((name)::text = 'discourse_automation_ids'::text)
```
[the remark this answers is not in the document]
Comment (tgxworld) [reply app/models/post_custom_field.rb:22]: Hmm that is odd. Can you check if we are adding annotations for plugins like the poll plugin?

[the remark this answers is not in the document]
Comment (OsamaSayegh) [reply app/models/post_custom_field.rb:22]: The poll plugin doesn't add indexes to core tables (e.g. `topic_custom_fields`) like automation does and neither do any of the existing core plugins.

File: plugins/automation/config/locales/client.en.yml [hunk 4]
```diff
@@ -21,10 +21,10 @@ en:
         confirm: "Are you sure you want to delete `%{name}`?"
       fields:
         key_value:
-          label:
-            zero: Configure
-            one: Edit Configuration (%{count})
-            other: Edit Configuration (%{count})
```
[the remark this answers is not in the document]
Comment (TaherAlhussein) [reply plugins/automation/config/locales/client.en.yml]: https://github.com/discourse/discourse/pull/26432/commits/51ea0d4ef00bb8285fe1de68f8128ea5ec95103d

File: script/i18n_lint.rb [hunk 5]
```diff
@@ -42,6 +42,10 @@ class LocaleFileValidator
   PLURALIZATION_KEYS = %w[zero one two few many other]
   ENGLISH_KEYS = %w[one other]
 
+  EXEMPTED_DOUBLE_CURLY_BRACKET_KEYS = [
+    "js.discourse_automation.scriptables.auto_responder.fields.word_answer_list.description",
+  ]
```
[the remark this answers is not in the document]
Comment (TaherAlhussein) [reply script/i18n_lint.rb:47]: https://github.com/discourse/discourse/pull/26432/commits/fa13f1e51dc9d819cd756387da8cb96e785b0280

File: app/models/post_custom_field.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (TaherAlhussein) [reply app/models/post_custom_field.rb:22]: https://github.com/discourse/discourse/pull/26432/commits/ae8de204ae1f84d068e2858c4b6b0f24235566bb

File: app/models/topic_custom_field.rb (hunk 1 above)
Comment (TaherAlhussein) [reply app/models/topic_custom_field.rb:23]: https://github.com/discourse/discourse/pull/26432/commits/ae8de204ae1f84d068e2858c4b6b0f24235566bb

Comment (TaherAlhussein) [reply app/models/topic_custom_field.rb:23]: https://github.com/discourse/discourse/pull/26432/files#r1548913457

File: app/models/post_custom_field.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (TaherAlhussein) [reply app/models/post_custom_field.rb:22]: https://github.com/discourse/discourse/pull/26432/files#r1548913457

**Pull request merged**